### PR TITLE
Add a configuration for the rgbkb Zen1

### PR DESCRIPTION
**Pull request by jpotier**
## Description

New keymap providing an example of qwerty laid out in a BÉPO friendly way.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/rgbkb/zen/rev1/keymaps/samae/keymap.c [hunk 1]
```diff
@@ -0,0 +1,89 @@
+#include QMK_KEYBOARD_H
+
+extern keymap_config_t keymap_config;
```
Comment (drashna) [inline keyboards/rgbkb/zen/rev1/keymaps/samae/keymap.c:3]: This isn't needed, actually.
```suggestion
```

File: keyboards/rgbkb/zen/rev1/keymaps/samae/keymap.c [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+enum layer_number {
+    _QWERTY = 0,
```
Comment (drashna) [inline keyboards/rgbkb/zen/rev1/keymaps/samae/keymap.c:10]: You don't need 0 explicitly defined here. But either way is just fine. ```suggestion
    _QWERTY,
```

File: keyboards/rgbkb/zen/rev1/keymaps/samae/keymap.c [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+   * |------+------+------+------+------+------+------.              .--------+------+------+------+------+------+------|
+   * | Ctrl | GUI  | Alt  |RGBTOG| NAV  |Space | ESC  |              |Enter|Space/NAV| AltGr|  NAV |      |   -  |   =  |
+   * `------------------------------------------------'              '--------------------------------------------------'
+   */
+  [_QWERTY] = LAYOUT( \
+    KC_GRV,   KC_1,    KC_2,    KC_3,    KC_4,     KC_5,                      KC_6,               KC_7,     KC_8,    KC_9,    KC_0,    KC_BSPC, \
+    KC_TAB,   KC_Q,    KC_W,    KC_E,    KC_R,     KC_T,                      KC_Y,               KC_U,     KC_I,    KC_O,    KC_P,    KC_BSLS, \
+    KC_DEL,   KC_A,    KC_S,    KC_D,    KC_F,     KC_G,                      KC_H,               KC_J,     KC_K,    KC_L,    KC_SCLN, KC_QUOT, \
+    KC_LSFT,  KC_Z,    KC_X,    KC_C,    KC_V,     KC_B,                      KC_N,               KC_M,     KC_COMM, KC_DOT,  KC_SLSH, RGB_MOD, \
+    KC_LCTL,  KC_LGUI, KC_LALT, RGB_TOG, MO(_NAV), KC_SPC, KC_GESC,  KC_ENT,  LT(_NAV, KC_SPACE), KC_RALT, MO(_NAV), _______, KC_MINS, KC_EQL   \
```
Comment (drashna) [inline keyboards/rgbkb/zen/rev1/keymaps/samae/keymap.c:41]:
```suggestion
  [_QWERTY] = LAYOUT( 
    KC_GRV,   KC_1,    KC_2,    KC_3,    KC_4,     KC_5,                      KC_6,               KC_7,     KC_8,    KC_9,    KC_0,    KC_BSPC,
    KC_TAB,   KC_Q,    KC_W,    KC_E,    KC_R,     KC_T,                      KC_Y,               KC_U,     KC_I,    KC_O,    KC_P,    KC_BSLS,
    KC_DEL,   KC_A,    KC_S,    KC_D,    KC_F,     KC_G,                      KC_H,               KC_J,     KC_K,    KC_L,    KC_SCLN, KC_QUOT,
    KC_LSFT,  KC_Z,    KC_X,    KC_C,    KC_V,     KC_B,                      KC_N,               KC_M,     KC_COMM, KC_DOT,  KC_SLSH, RGB_MOD,
    KC_LCTL,  KC_LGUI, KC_LALT, RGB_TOG, MO(_NAV), KC_SPC, KC_GESC,  KC_ENT,  LT(_NAV, KC_SPACE), KC_RALT, MO(_NAV), _______, KC_MINS, KC_EQL  
```

File: keyboards/rgbkb/zen/rev1/keymaps/samae/keymap.c [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+   * |------+------+------+------+------+------+------.              .------+------+------+------+------+------+------|
+   * | Ctrl | GUI  | Alt  |RGBMOD|      |      |      |              | RESET|      |      |      | Mute | VOLUP| VOLDN|
+   * `------------------------------------------------'              '------------------------------------------------'
+   */
+  [_NAV] = LAYOUT( \
+    KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,                     KC_F7,   KC_F8,    KC_F9,   KC_F10,  KC_F11,  KC_F12,  \
+    _______, RGB_SAI, RGB_VAI, RGB_SAD, _______, KC_LBRC,                   KC_RBRC, KC_PGUP,  KC_UP,   KC_PGDN, KC_INS,  KC_HOME, \
+    _______, RGB_HUD, RGB_VAD, RGB_HUI, _______, _______,                   _______, KC_LEFT,  KC_DOWN, KC_RGHT, KC_DEL,  KC_END,  \
+    KC_LSFT, _______, _______, _______, _______, _______,                   _______, _______,  _______, KC_MPLY, KC_MPRV, KC_MNXT, \
+    KC_LCTL, KC_LGUI, KC_LALT, RGB_MOD, _______, _______, _______, RESET,   _______, _______,  _______, KC_MUTE, KC_VOLU, KC_VOLD  \
```
Comment (drashna) [inline keyboards/rgbkb/zen/rev1/keymaps/samae/keymap.c:62]:
```suggestion
  [_NAV] = LAYOUT( 
    KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,                     KC_F7,   KC_F8,    KC_F9,   KC_F10,  KC_F11,  KC_F12, 
    _______, RGB_SAI, RGB_VAI, RGB_SAD, _______, KC_LBRC,                   KC_RBRC, KC_PGUP,  KC_UP,   KC_PGDN, KC_INS,  KC_HOME,
    _______, RGB_HUD, RGB_VAD, RGB_HUI, _______, _______,                   _______, KC_LEFT,  KC_DOWN, KC_RGHT, KC_DEL,  KC_END, 
    KC_LSFT, _______, _______, _______, _______, _______,                   _______, _______,  _______, KC_MPLY, KC_MPRV, KC_MNXT,
    KC_LCTL, KC_LGUI, KC_LALT, RGB_MOD, _______, _______, _______, RESET,   _______, _______,  _______, KC_MUTE, KC_VOLU, KC_VOLD 
```

File: keyboards/rgbkb/zen/rev1/keymaps/samae/keymap.c [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        #ifdef AUDIO_ENABLE
+          PLAY_NOTE_ARRAY(tone_qwerty, false, 0);
+        #endif
+        persistant_default_layer_set(1UL<<_QWERTY);
+      }
+      return false;
+      break;
+  }
+  return true;
+}
```
Comment (drashna) [inline keyboards/rgbkb/zen/rev1/keymaps/samae/keymap.c:89]: None of this is needed, or even used, really. The default layer function is is depricated, as well. 
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/rgbkb/zen/rev1/keymaps/samae/keymap.c (hunk 2 above)
Comment (jpotier) [reply keyboards/rgbkb/zen/rev1/keymaps/samae/keymap.c:9]: I'm not bothered too much by being explicit here. If it's ok with  you I'd rather keep it.

File: keyboards/rgbkb/zen/rev1/keymaps/samae/keymap.c (hunk 3 above)
Comment (jpotier) [reply keyboards/rgbkb/zen/rev1/keymaps/samae/keymap.c]: Interesting!

File: keyboards/rgbkb/zen/rev1/keymaps/samae/keymap.c (hunk 5 above)
Comment (jpotier) [reply keyboards/rgbkb/zen/rev1/keymaps/samae/keymap.c:88]: Thanks! This piece of config has been in limbo for a year and some, I haven't kept up with the changes nearly enough.

File: keyboards/rgbkb/zen/rev1/keymaps/samae/keymap.c [hunk 6]
```diff
@@ -0,0 +1,64 @@
+#include QMK_KEYBOARD_H
```
Comment (noroadsleft) [inline keyboards/rgbkb/zen/rev1/keymaps/samae/keymap.c:1]: Could you add a license header here, please?

Here's a sample one to modify:

https://github.com/qmk/qmk_firmware/blob/940280c43594833e2208d8514452b8ab88c14a9c/quantum/template/base/keymaps/default/keymap.c#L1-L15

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/rgbkb/zen/rev1/keymaps/samae/keymap.c (hunk 6 above)
Comment (jpotier) [reply keyboards/rgbkb/zen/rev1/keymaps/samae/keymap.c:1]: Sure thing (been a busy time these last days, sorry for the delay)

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

**Review by fauxpark — APPROVED** [review]

**Comment by noroadsleft** [comment]
For future reference, we recommend against committing to your `master` branch as you've done here, because pull requests from modified `master` branches can make it more difficult to keep your QMK fork updated. It is highly recommended for QMK development – regardless of what is being done or where – to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your fork updated here:

[**Best Practices: Your Fork's Master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

[Fixing Your Branch](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) will walk you through fixing up your `master` branch moving forward. If you need any help with this just ask.

Thanks for contributing!

**Comment by jpotier** [comment]
Thanks, I'll change this for next time.

**Pull request merged**